src/librawspeed/decoders/IiqDecoder.cpp

```diff
@@ -242,6 +260,153 @@ void IiqDecoder::DecodePhaseOneC(const std::vector<IiqStrip>& strips,
     DecodeStrip(strip, width, height);
 }
 
+void IiqDecoder::CorrectPhaseOneC(ByteStream meta_data, uint32 split_row,
+                                  uint32 split_col) {
+  meta_data.skipBytes(8);
+  const uint32 bytes_to_entries = meta_data.getU32();
+  meta_data.setPosition(bytes_to_entries);
+  const uint32 entries_count = meta_data.getU32();
+  meta_data.skipBytes(4);
+
+  // this is how much is to be read for all the entries
+  ByteStream entries(meta_data.getStream(entries_count, 12));
+  meta_data.setPosition(0);
+
+  bool quadrant_multiplier_applied = false;
+
+  for (uint32 entry = 0; entry < entries_count; entry++) {
+    const uint32 tag = entries.getU32();
+    const uint32 len = entries.getU32();
+    const uint32 offset = entries.getU32();
+
+    switch (tag) {
+    case 0x431:
+      if (!quadrant_multiplier_applied) {
+        CorrectQuadrantMultipliersCombined(meta_data.getSubStream(offset, len),
+                                           split_row, split_col);
+      }
+      quadrant_multiplier_applied = true;
+      break;
+    default:
+      break;
+    }
+  }
+}
+
+// This function defines a correction that compensates for the fact
+// that IIQ files may come from a camera with multiple (four, in this
+// case) sensors combined into a single "sensor."  Because the
+// different sensors may have slightly different responses, we need to
+// multiply the pixels in each by a correction factor to ensure that
+// they blend together smoothly.  The correction factor is not a
+// single multiplier, but a curve defined by seven control points.
+// Each curve's control points share the same seven X-coordinates.
+void IiqDecoder::CorrectQuadrantMultipliersCombined(ByteStream data,
+                                                    uint32 split_row,
+                                                    uint32 split_col) {
+
+  // The curves should all include (0, 0) and (65535, 65535), so the
+  // first and last points are predefined and the middle seven are
+  // read from the file
+  uint32 shared_x_coords[9] = {0, 0, 0, 0, 0, 0, 0, 0, 65535};
+  for (int i = 0; i < 7; i++)
+    shared_x_coords[i + 1] = data.getU32();
```

Comment:
```c++
std::generate_n(std::next(shared_x_coords.begin(), 1), 7,
                [&data]() { return data.getU32(); });
```

Reply:
Damn, I guess I can't skip learning the new C++ ;)